cmd/aws-s3-proxy/pkg/serve.go

```diff
@@ -28,112 +29,175 @@ var serveCmd = &cobra.Command{
+
+	serveCmd.Flags().StringP("upstream-bucket", "", "", "upstream s3 bucket")
+	viperBindFlag("s3bucket", serveCmd.Flags().Lookup("upstream-bucket"))
+
+	serveCmd.Flags().StringP("upstream-key-prefix", "", "", "upstream s3 path/key prefix")
+	viperBindFlag("s3prefix", serveCmd.Flags().Lookup("upstream-key-prefix"))
+
+	serveCmd.Flags().StringP("ssl-cert-path", "", "", "path to ssl cert")
+	viperBindFlag("sslcert", serveCmd.Flags().Lookup("ssl-cert-path"))
+
+	serveCmd.Flags().StringP("ssl-key-path", "", "", "path to ssl key")
+	viperBindFlag("sslkey", serveCmd.Flags().Lookup("ssl-key-path"))
+
+	serveCmd.Flags().StringP("strip-path", "", "", "strip path prefix")
+	viperBindFlag("strippath", serveCmd.Flags().Lookup("strip-path"))
+
+	if err := serveCmd.MarkFlagRequired("upstream-bucket"); err != nil {
+		logger.Fatal(err)
+	}
+
+	if len(os.Getenv("S3_PROXY_BASIC_AUTH_PASS")) != 0 {
+		viper.Set("basicauthpass", os.Getenv("S3_PROXY_BASIC_AUTH_PASS"))
+	}
 
 	// Configs that need transformation
-	serveCmd.PersistentFlags().BoolVar(&directoryListingFormat, "directory-listing-format", false, "toggle directory listing spider formatted")
+	serveCmd.Flags().BoolVar(&directoryListingFormat, "directory-listing-format", false, "toggle directory listing spider formatted")
 
 	if directoryListingFormat {
-		c.DirListingFormat = "html"
+		viper.Set("directorylistingformat", "html")
 	}
 
-	serveCmd.PersistentFlags().IntVar(&idleConnTimeout, "idle-connection-timeout", 10, "idle connection timeout in seconds")
-	c.IdleConnTimeout = time.Duration(idleConnTimeout) * time.Second
+	serveCmd.Flags().IntP("idle-connection-timeout", "", 10, "idle connection timeout in seconds")
+	viper.Set("idleconntimeout", time.Duration(idleConnTimeout)*time.Second)
 
-	serveCmd.PersistentFlags().IntVar(&guessBucketTimeout, "guess-bucket-timeout", 10, "timeout, in seconds, for guessing bucket region")
-	c.GuessBucketTimeout = time.Duration(guessBucketTimeout) * time.Second
+	serveCmd.Flags().IntP("guess-bucket-timeout", "", 10, "timeout, in seconds, for guessing bucket region")
+	viper.Set("guessbuckettimeout", time.Duration(guessBucketTimeout)*time.Second)
 
 	// Configs with default AWS overrides
-	serveCmd.PersistentFlags().StringVar(&c.AwsAPIEndpoint, "aws-api-endpoint", "", "AWS API Endpoint")
+	serveCmd.Flags().StringP("aws-api-endpoint", "", "", "AWS API Endpoint")
+	viperBindFlag("awsapiendpoint", serveCmd.Flags().Lookup("aws-api-endpoint"))
 
 	if len(os.Getenv("AWS_API_ENDPOINT")) != 0 {
-		c.AwsAPIEndpoint = os.Getenv("AWS_API_ENDPOINT")
+		viper.Set("awsapiendpoint", os.Getenv("AWS_API_ENDPOINT"))
 	}
 
-	serveCmd.PersistentFlags().StringVar(&c.AwsRegion, "aws-region", "us-east-1", "AWS region for s3, default AWS env vars will override")
+	serveCmd.Flags().StringP("aws-region", "", "us-east-1", "AWS region for s3, default AWS env vars will override")
+	viperBindFlag("awsregion", serveCmd.Flags().Lookup("aws-region"))
 
 	if len(os.Getenv("AWS_REGION")) != 0 {
```

Comment:
Why aren't these using the `S3_PROXY` prefix and using `AutomaticEnv`?

Reply:
This is overriding if the default AWS env vars are present.

Reply:
Since this repo is public I want to preserve expected behavior.